HeatStroke/Services/IO/JoystickInputBuffer.cpp

```diff
@@ -0,0 +1,237 @@
+//====================================================================================
+// JoystickInputBuffer
+// Author: Bradley Cooper
+//
+// A resource handler for joystick (gamepad) input.
+//====================================================================================
+
+#include "JoystickInputBuffer.h"
+
+#include <cassert>
+
+namespace HeatStroke
+{
+	// Static singleton instance
+	JoystickInputBuffer* JoystickInputBuffer::s_pJoystickInputBufferInstance = nullptr;
```

Comment:
Am I missing something, or does making this a singleton mean that if PlayerA and PlayerB are both using a gamepad, they're essentially controlling each others game?